### PR TITLE
[To rel/0.13][IOTDB-5133] Export schema tool

### PR DESCRIPTION
**Pull request by Cpaulyz**
## Description

Add an export schema function to export MTree and Tag from the current system into a non-deleted mlog.bin for loading mlog in a non-empty iotdb correctly.

The schema export operation exports the information about store group, timeseries, and schema template in the current IoTDB in the form of `mlog.bin` and `tlog.txt` to the specified directory.

The exported `mlog.bin` and `tlog.txt` files can be incrementally loaded into an IoTDB instance.

### Export Schema SQL

```
EXPORT SCHEMA '<path/dir>' 
```

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 1]
```diff
@@ -1336,6 +1339,81 @@ public Pair<List<MeasurementPath>, Integer> getMeasurementPathsWithAlias(
...
+        throw new IOException(String.format("%s is not a directory.", dir.getAbsolutePath()));
+      }
+    }
+    Files.deleteIfExists(
+        FileSystems.getDefault()
+            .getPath(
+                new File(dir.getAbsolutePath(), MetadataConstant.METADATA_LOG).getAbsolutePath()));
+    Files.deleteIfExists(
+        FileSystems.getDefault()
+            .getPath(new File(dir.getAbsolutePath(), MetadataConstant.TAG_LOG).getAbsolutePath()));
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:1356]: It's better to remind the user there's existing files and let the user delete them.

File: server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTree.java [hunk 2]
```diff
@@ -230,6 +234,139 @@ public void serializeTo(String snapshotPath) throws IOException {
...
+                mLogWriter.setSchemaTemplate(
+                    new SetTemplatePlan(
+                        node.getSchemaTemplate().getName(),
+                        getCurrentPartialPath(node).getFullPath()));
+                if (node.isUseTemplate()) {
+                  mLogWriter.setUsingSchemaTemplate(getCurrentPartialPath(node));
+                }
+              } catch (IOException e) {
+                logger.error(e.getMessage());
+              }
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTree.java:258]: Extract the common code.

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 3]
```diff
@@ -1336,6 +1339,81 @@ public Pair<List<MeasurementPath>, Integer> getMeasurementPathsWithAlias(
...
+        }
+      }
+      // export timeseries
+      mtree.exportSchema(mLogWriter);
+      // export tag
+      File tagTargetFile = new File(dir, MetadataConstant.TAG_LOG);
+      File tagSourceFile = new File(config.getSchemaDir(), MetadataConstant.TAG_LOG);
+      Path sourcePath = FileSystems.getDefault().getPath(tagSourceFile.getAbsolutePath());
+      Path targetPath = FileSystems.getDefault().getPath(tagTargetFile.getAbsolutePath());
+      Files.createLink(targetPath, sourcePath);
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:1413]: If cross disk, this operation may fail.

**Review by MarcosZyk — COMMENTED** [review]
PTAL

File: server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTree.java (hunk 2 above)
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTree.java]: done

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java (hunk 3 above)
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java]: replaced by `Files.copyFile`

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java (hunk 1 above)
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java]: Done. Add test in `testExportSchemaFail`

**Review by MarcosZyk — APPROVED** [review]

**Pull request merged**